data/src/main/java/org/oppia/android/data/backends/gae/RemoteAuthNetworkInterceptor.kt

```diff
@@ -0,0 +1,48 @@
+package org.oppia.android.data.backends.gae
+
+import android.content.Context
+import okhttp3.Interceptor
+import okhttp3.Request
+import okhttp3.Response
+import org.oppia.android.app.utility.getVersionCode
+import org.oppia.android.app.utility.getVersionName
+import java.io.IOException
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Interceptor on top of Retrofit to modify outgoing requests for authenticating messages.
+ *
+ * The Interceptor adds header parameters to outgoing messages.
+ */
+@Singleton
```

Comment:
This class is stateless so it shouldn't be scoped.

Reply:
Done